lib/monitoring/headend.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Monitoring.Headend do
+  require Logger
+
+  def update_active_headend_ip(ip) do
+    valid? = ip == "172.20.145.20" or ip == "172.20.145.22"
+    change? = ip != Application.get_env(:realtime_signs, :sign_head_end_host)
+
+    cond do
+      not valid? ->
+        Logger.warn("active_headend_ip: invalid ip provided: #{ip}")
+        {:bad_request, "bad request"}
+
+      change? ->
+        with {:ok, _} = response <- ExternalConfig.S3.put_active_headend_ip(ip) do
```

Comment:
This should read the `external_config_getter` config value rather than assuming `ExternalConfig.S3`

Reply:
ahh great catch, thank you!